### ISSUE TITLE
Glob does not work correctly

### ISSUE THREAD

**Issue by dr-vortex** [google-github-actions/upload-cloud-storage#337]
### TL;DR

Using a glob like `./dir/my-app*` yields `Error: google-github-actions/upload-cloud-storage failed with: ENOENT: no such file or directory, lstat '/home/runner/work/repo/repo/dir/my-app*'`, even though `dir/my-app.exe` exists.

### Expected behavior

The glob pattern to resolve into the files it matches

### Observed behavior

It treated the glob pattern like a file, without resolving it first

### Action YAML

```yaml
name: Continuous Integration

on:
    push:
        branches:
            - main

jobs:
    ci:
        runs-on: ${{ matrix.os }}
        strategy:
            matrix:
                os: [ubuntu-latest, macos-latest, windows-latest]
        name: ${{ matrix.os }}
        permissions:
            contents: read
            id-token: write
        defaults:
            run:
                shell: bash
        steps:
            - name: Checkout
              uses: actions/checkout@v3

            - name: Set up Node.js
              uses: actions/setup-node@v3
              with:
                  node-version: 18

            - name: Install dependencies
              run: npm install

            - name: Formatting
              run: npm run format:check

            - name: Linting
              run: npm run lint

            # Once tests are added
            #-name: Test
            # run: npm run test

            - name: Build client
              run: npm run build:client

            - name: Build server
              run: npm run build:server

            - name: Authenticate to Google Cloud
              uses: google-github-actions/auth@v1
              with:
                  workload_identity_provider: projects/195808230594/locations/global/workloadIdentityPools/github/providers/oidc
                  service_account: github-actions@blankstorm.iam.gserviceaccount.com

            - name: Upload build
              uses: google-github-actions/upload-cloud-storage@v1
              with:
                  path: ./dist/blankstorm-client*
                  destination: drvortex.dev
                  parent: false
```

### Log output

https://github.com/dr-vortex/blankstorm/actions/runs/5700838637/job/15451017276

**Comment by sethvargo** [comment]
Hi @dr-vortex - please share your [debug output](https://docs.github.com/en/actions/monitoring-and-troubleshooting-workflows/enabling-debug-logging). Also, it appears you are running this on 3 OSes (Linux, Mac, Windows); is that intended? At present, that matrix runs in parallel and all 3 OSes are going to try and upload different files to the same bucket and paths.

**Comment by dr-vortex** [comment]
Here is the run with debug logging: https://github.com/dr-vortex/blankstorm/actions/runs/5700838637/job/15451347134

> Also, it appears you are running this on 3 OSes (Linux, Mac, Windows); is that intended? At present, that matrix runs in parallel and all 3 OSes are going to try and upload different files to the same bucket and paths.

Yes, I run tests on each platform, which uploads files specific to that platform.

**Comment by sethvargo** [comment]
Hi @dr-vortex - that's the runner logs. I need the [debug logs](https://docs.github.com/en/actions/monitoring-and-troubleshooting-workflows/enabling-debug-logging#enabling-step-debug-logging), which requires setting a secret of `ACTIONS_STEP_DEBUG` to `true` and re-running the job. You should a LOT of output including "Computed absoluteRoot from" and "Found XX files".

> Yes, I run tests on each platform, which uploads files specific to that platform.

Your current upload action has no namespace by matrix. That means files that are generated that have the same name will be overwritten in a non-deterministic way (depending on which runner runs first):

```yaml
- name: Upload build
  uses: google-github-actions/upload-cloud-storage@v1
    with:
      path: ./dist/blankstorm-client*
      destination: drvortex.dev
      parent: false
```

I'd probably expect something like this:

```yaml
- name: Upload build
  uses: google-github-actions/upload-cloud-storage@v1
    with:
      path: ./dist/blankstorm-client*
      destination: drvortex.dev/${matrix.os}
      parent: false
```

**Comment by dr-vortex** [comment]
@sethvargo I apologize for the inconvenience, I skimmed the article and set `ACTIONS_RUNNER_DEBUG ` to true (the wrong variable). I set the correct variable and reran [the tests](https://github.com/dr-vortex/blankstorm/actions/runs/5701374199/job/15452029658). 

> Your current upload action has no namespace by matrix. That means files that are generated that have the same name will be overwritten in a non-deterministic way (depending on which runner runs first):

That is correct. Each platform generates different files (i.e. `blankstorm-client-<verison>.exe`, `blankstorm-client-<verison>-linux`, and `blankstorm-client-<verison>-macos`). The server executables are overwritten since `pkg` cross-compiles, though they are all the same.

**Comment by sethvargo** [comment]
Hi @dr-vortex sorry, but I'm still not setting the debug output in that link. It looks like it's still the runner debug output. There should be a lot of output from the upload-cloud-storage action (like hundreds of lines).

**Comment by dr-vortex** [comment]
That's odd. I have the repository variable `ACTIONS_STEP_DEBUG` set to true. Does it need to be a secret?

**Comment by dr-vortex** [comment]
![image](https://github.com/google-github-actions/upload-cloud-storage/assets/75621402/44064d27-323f-49fc-8838-30b9346631ac)
![image](https://github.com/google-github-actions/upload-cloud-storage/assets/75621402/11d32657-1066-4ed7-a80f-679c8ff4be92)
![image](https://github.com/google-github-actions/upload-cloud-storage/assets/75621402/d5892677-2d88-4768-864a-1b060719b644)

I also downloaded the full logs and they do not have "like hundreds of lines".

**Comment by sethvargo** [comment]
Hi @dr-vortex - I'm not sure. Is the value `true` (with no leading or trailing space)? I just verified that debug logging works with ACTIONS_STEP_DEBUG set as a secret on our repo. Here's the kind of log output I'd expect to see: https://github.com/google-github-actions/upload-cloud-storage/actions/runs/5705589160/job/15460424392

In "Upload files", you'll see output like:

```text
##[debug]Computed computedGlob from "" to ""
##[debug]Found 2 files: ["test1.txt","test2.txt"]
##[debug]Processing gcloudignore
##[debug]Uploading 2 files: ["test1.txt","test2.txt"]
##[debug]Computed bucket as "gh-upload-cloud-storage-24c0"
##[debug]Computed prefix as "testprefix"
```

**Comment by dr-vortex** [comment]
Confirmed that there is not extra whitespace. I'm not sure why this issue is occurring. Perhaps it has something to do with how external actions work with a repository's secrets and variables? I'm just throwing an idea out.

**Comment by sethvargo** [comment]
Okay - I think I see the problem. We're never making it past parsing the `path` input to _see_ any debug logs. The `path` attribute must be a real filepath on disk. You can specify the additional `glob` attribute separately: https://github.com/google-github-actions/upload-cloud-storage#glob-pattern

**Comment by dr-vortex** [comment]
@sethvargo Ah, yes:

![image](https://github.com/google-github-actions/upload-cloud-storage/assets/75621402/19188ce3-5671-4623-92c8-02710f278f55)
Clearly this is not the path property. Thank you for the help!

**Comment by dr-vortex** [comment]
I'm now getting `Error: google-github-actions/upload-cloud-storage failed with: error code unauthorized_client: Could not obtain a value for google.subject from the given credential.` Which attribute mapping should I use? I currently have `google.subject` mapping to `attribute.repository_owner`.

**Comment by sethvargo** [comment]
Hi @dr-vortex - that's a better question for a separate issue on github.com/google-github-actions/auth. Please see the [troubleshooting guide](https://github.com/google-github-actions/auth/blob/main/docs/TROUBLESHOOTING.md) there too.

**Comment by james-pre** [comment]
@sethvargo I've updated my GitHub username from @dr-vortex to @james-pre. Please update your comment(s), and I can delete this one. I'm writing this comment when I can't make the change myself.
